### PR TITLE
SISRP-24217 Switch to LDAP as preferred source for bulk attributes

### PR DESCRIPTION
**Pull request by raydavis**
https://jira.berkeley.edu/browse/SISRP-24217

We already prefer LDAP for the more detailed single-person attributes.

### CODE REVIEW

File: app/models/user/basic_attributes.rb [hunk 1]
```diff
@@ -6,14 +6,19 @@ def attributes_for_uids(uids)
       return [] if uids.blank?
       uid_set = uids.to_set
       attrs = []
-      # Oracle dislikes ' IN ()' queries with more than 1000 items.
-      uids.each_slice(1000) do |next_batch|
-        CampusOracle::Queries.get_basic_people_attributes(next_batch).each do |result|
-          uid_set.delete result['ldap_uid']
-          attrs << transform_campus_row(result)
+      # Primary source for bulk attributes is CalNet LDAP.
+      attrs.concat CalnetLdap::UserAttributes.get_bulk_attributes(uid_set)
```
Comment (paulkerschen) [inline app/models/user/basic_attributes.rb]: @raydavis Minor, but at this point might we as well get rid of the empty-array initialization and just write `attrs = CalnetLdap::UserAttributes.get_bulk_attributes(uid_set)`?

Comment (paulkerschen) [inline app/models/user/basic_attributes.rb]: @raydavis Also looks like `CalnetLdap::UserAttributes.get_bulk_attributes` has a chance of returning nil, so we should probably fall back to an empty array either here or in that method.

**Comment by paulkerschen** [comment]
Looks good; a couple of minor comments. Also, with bulk LDAP now on the front lines, might be time to pursue this TODO through queries and/or experimental benchmarking of our own:
https://github.com/ets-berkeley-edu/calcentral/blob/master/app/models/calnet_ldap/client.rb#L12

**Comment by raydavis** [comment]
Thank you for the very helpful review, @paulkerschen -- amendment & CalNet query coming.

**Comment by raydavis** [comment]
Have to close this while we sort out performance issues.

**Pull request closed without merging**